pkg/lib/oauth/handler/handler_token.go

```diff
@@ -597,6 +597,9 @@ func (h *TokenHandler) handleAppInitiatedSSOToWebToken(
 	if err != nil {
 		return nil, protocol.NewError("invalid_request", "invalid subject_token")
 	}
+	if idToken.Issuer() != r.Audience() {
+		return nil, protocol.NewError("invalid_request", "invalid audience")
```

Comment:
Actually, the real logic here should be "expected audience to be {authorzation server}". I just used the issuer here to obtain the value representing authorization server.

Reply:
Isn't the iss of the ID token be {authorization server}?

Reply:
Yes. But if I mention that in the error message, looks like we are checking the value with a wrong place (Because audience is related to {authorization server} not {id_token.issuer}).

Reply:
What about we follow pkg/lib/oauth/oidc/metadata.go:51 and use p.Endpoints.Origin().String() ?

and the error message would be `fmt.Sprintf("expected audience to be %v", p.Endpoints.Origin().String())`

Reply:
Maybe what I should do is to change the checking to check against the server endpoint but not the id token issuer.